onlineconf.go

```diff
@@ -51,53 +94,58 @@ func init() {
 				}
 
 			case err := <-watcher.Errors:
-				log.Printf("Watch %v error: %v\n", configDir, err)
+				log.Printf("Watch %v error: %v\n", ConfigDir, err)
 			}
 		}
 	}()
 }
 
-// SetOutput sets the output destination for the standard logger.
-func SetOutput(w io.Writer) {
-	log.SetOutput(w)
-}
+func (m *Module) reopen() error {
+	log.Printf("Reopen %s\n", m.filename)
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
 
-type Module struct {
-	m    sync.RWMutex
-	name string
-	file string
-	cdb  *cdb.Cdb
-}
+	oldMmappedFile := m.mmappedFile
+	filename := m.filename
 
-func newModule(name string) *Module {
-	file := fmt.Sprintf("%s/%s.cdb", configDir, name)
-	cdb, err := cdb.Open(file)
+	file, err := os.Open(filename)
 	if err != nil {
-		panic(err)
+		return err
 	}
-	return &Module{name: name, file: file, cdb: cdb}
-}
+	defer file.Close()
 
-func (m *Module) reopen() {
-	log.Printf("Reopen %s\n", m.file)
-	m.m.Lock()
-	defer m.m.Unlock()
-	cdb, err := cdb.Open(m.file)
+	fileInfo, err := file.Stat()
 	if err != nil {
-		log.Printf("Reopen file %v error: %v\n", m.file, err)
-	} else {
-		m.cdb.Close()
-		m.cdb = cdb
+		return err
+	}
+
+	mmappedFile, err := mmap.NewSharedFileMmap(file, 0, int(fileInfo.Size()), syscall.PROT_READ)
```

Comment:
I am really surprised what none of three golang CDB reader implementations use `mmap` under the hood.
Good catch!